samples/open-telemetry/customizing/sample.md

```diff
@@ -0,0 +1,94 @@
+---
+title: Customizing OpenTelemetry tracing
+summary: Demonstrates how to add data to existing OpenTelemetry traces
+reviewed: 2022-07-04
+component: Core
+---
+
+This sample shows how to extend the OpenTelemetry activities in a variety of ways.
+
+## Running the project
+
+The code consists of a single endpoint project that sends messages to itself.
+
+Press `O` to send a `CreateOrder` message with a randomized `OrderId`. When the message is handled, two more messages are created (`BillOrder` and `ShipOrder`).
+
+As the messages are sent and processed, trace data is exported to the console. Some of the trace data originates from NServiceBus and some from custom activities in the sample.
+
+## Code walk through
+
+### Global configuration
+
+OpenTelemetry is configured to export all traces to the command line. It includes the `NServiceBus.Core` source which is built into NServiceBus and a custom activity source defined in the sample (see below).
+
+snippet: open-telemetry-config
+
+A custom processor is registered which adds the machine name as a tag to every activity created by this trace listener.
+
+snippet: custom-processor
+
+### Custom activities
+
+The sample includes a custom activity source.
+
+snippet: custom-activity-source
+
+The handler for `CreateOrder` includes a custom activity that wraps around the billing section.
+
+snippet: custom-activity-in-handler
+
+This will automatically be created as a child activity of the invoke handler activity created by NServiceBus. The NServiceBus send message activity will treat this custom activity as it's parent.
+
+```
```

Comment:
Are these mermaid diagrams? Wouldn't the snippet require `mermaid` after the backticks to render properly?

Reply:
They aren't mermaid diagrams. They just use the codeblock style to preserve spaces. I did mess with trying to get these to render with mermaid using the gannt view but ultimately gave it up as not being that much of an improvement in readibility, and a whole lot harder to maintain.